### PR TITLE
Added benchmark goal and other clarifications to README

### PR DESCRIPTION
**Pull request by LesnyRumcajs**
https://github.com/LesnyRumcajs/grpc_bench/issues/81

### CODE REVIEW

**Comment by LesnyRumcajs** [comment]
It's only a draft, let's discuss if anything is not right / unclear. Or you can't possibly imagine anyone finding beauty in Ruby (which will hurt my feelings). :wink:

File: README.md [hunk 1]
```diff
@@ -4,6 +4,30 @@ One repo to finally have a clear, objective gRPC benchmark with code for everyon
...
+
+We try to provide some metrics to make this decision easier:
+
+* req/s - the number of requests the service was able to successfully serve
+* average latency, and 90/95/99 percentiles - time from sending a request to receiving the response
+* average CPU, memory - average resource usage during the benchmark, as reported by `docker stats`
+
+# What this benchmark does NOT take into account
+
+1. Completness of the gRPC library. We test only basic unary RPC at the moment. This is the most common service method which may be enough for some business use cases, but not for the others. When you're happy about the results of some technology, you should check out it's documentation (if it exists) and decide yourself is it production-ready.
```
Comment (Trisfald) [inline README.md:27]: `Completeness`

Comment (Trisfald) [reply README.md:27]: `if is it production-ready`.. I think
also we can leave out the documentation remark - some motivated people would just go and read the code anyway :)

File: README.md [hunk 2]
```diff
@@ -4,6 +4,30 @@ One repo to finally have a clear, objective gRPC benchmark with code for everyon
...
+2. Taste. Some may find beauty in Ruby, some may feel like Java is the only real deal. Others treat languages as tools and don't care at all. We don't judge (officially 😉 ). Unless it's a huge state machine with raw `void` pointers. Ups!
```
Comment (Trisfald) [inline README.md:28]: Haha what if one finds beauty in a massive mosaic of void pointers?

Comment (LesnyRumcajs) [reply README.md:28]: Then I'll proudly put `I was scolded by Linus Torvalds himself` in my resumé!

**Review by Trisfald — APPROVED** [review]

**Comment by brunoborges** [comment]
I think it is important that the benchmark specifies how it should be run, otherwise results will be very distorted.

For example, this benchmark should run on servers with at least X CPUs and with the client running with at least X CPUs dedicated to it and not competing with CPUs dedicated to the benchs/servers.

This is a particular detail that I found was distorting a lot the results.

**Comment by LesnyRumcajs** [comment]
> I think it is important that the benchmark specifies how it should be run, otherwise results will be very distorted.
> 
> For example, this benchmark should run on servers with at least X CPUs and with the client running with at least X CPUs dedicated to it and not competing with CPUs dedicated to the benchs/servers.
> 
> This is a particular detail that I found was distorting a lot the results.

Yes, I've just added recommendations for some of the parameters. Thank you!

File: README.md [hunk 3]
```diff
@@ -51,6 +51,12 @@ The benchmark can be configured through the following environment variables:
 |GRPC_CLIENT_QPS|Rate limit, in queries per second (QPS).|0 (*unlimited*)|
 |GRPC_CLIENT_CPUS|Maximum number of cpus used by the client.|1|
 
+### Parameter recommendations
+* `GRPC_BENCHMARK_DURATION` should not be too small. Some implementations need a *warm-up* before achieving their optimal performance and most real-life gRPC services are expected to be long running processes. From what we measured, **300s** should be enough.
+* `GRPC_SERVER_CPUS` + `GRPC_CLIENT_CPUS` should not exceed total number of cores on the machine. The reason for this is that you don't want the `ghz` client to steal precious CPU cycles from the service under test. Keep in mind that having the `GRPC_CLIENT_CPUS` too low may not saturate the service in some of the more performant implementations. Also keep in mind limitting the number of `GRPC_SERVER_CPUS` to 1 will severely hamper the performance for some technologies - is running a service on 1 CPU your use case? It may be, but keep in mind eventual load balancer also incurs some costs.
```
Comment (Trisfald) [inline README.md:56]: `limiting`

File: README.md [hunk 4]
```diff
@@ -51,6 +51,12 @@ The benchmark can be configured through the following environment variables:
...
+
+Other parameters wil depend on your use-case. Choose wisely.
```
Comment (Trisfald) [inline README.md:58]: `will`

**Pull request merged**